src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/BambooMigrationService.java

```diff
@@ -0,0 +1,147 @@
+        // If there is a URL, the URL is the recipient. In that case we take the notification id from the edit button
+        Elements entries = notificationTableBody.select("tr");
+        List<Long> notificationIds = new ArrayList<>();
+        for (Element entry : entries) {
+            Elements columns = entry.select("td");
+            if (columns.size() != 3) {
+                continue;
+            }
+            String recipient = columns.get(1).text();
+            String actions = columns.get(2).toString();
+            Pattern editNotificationIdPattern = Pattern.compile(".*?id=\"editNotification:(\\d+)\".*?");
+            if (recipient.trim().startsWith(artemisServerUrl)) {
+                Matcher matcher = editNotificationIdPattern.matcher(actions);
+                if (matcher.find()) {
+                    String notificationIdString = matcher.group(1);
+                    notificationIds.add(Long.parseLong(notificationIdString));
+                }
+            }
+        }
+
+        return notificationIds;
+    }
+
+    /**
+     * Deletes the given notification id for the given build plan
+     * Bamboo doesn't provide a REST endpoint for this action, but as we don't need to retrieve anything, it works the same way.
+     *
+     * @param buildPlanKey         The key of the build plan, which is usually the name combined with the project, e.g. 'EIST16W1-GA56HUR'.
+     * @param serverNotificationId the id of the notification to delete
+     */
+    private void deleteBuildPlanServerNotificationId(String buildPlanKey, Long serverNotificationId) {
+        MultiValueMap<String, String> parameters = new LinkedMultiValueMap<>();
+        parameters.add("buildKey", buildPlanKey);
+        parameters.add("notificationId", serverNotificationId.toString());
+
+        String requestUrl = bambooServerUrl + "/chain/admin/config/deleteChainNotification.action";
+        UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(requestUrl).queryParams(parameters);
+
+        restTemplate.exchange(builder.build().toUri(), HttpMethod.POST, null, String.class);
+    }
+
+    /**
+     * Creates a new notification for the given build plan
+     * Bamboo doesn't provide a REST endpoint for this action, but as we don't need to retrieve anything, it works the same way.
+     *
+     * @param buildPlanKey          The key of the build plan, which is usually the name combined with the project, e.g. 'EIST16W1-GA56HUR'.
+     * @param serverNotificationUrl the url of the endpoint to notify
+     */
+    private void createBuildPlanServerNotification(String buildPlanKey, String serverNotificationUrl) {
+        MultiValueMap<String, String> body = new LinkedMultiValueMap<>();
+        body.add("conditionKey", "com.atlassian.bamboo.plugin.system.notifications:chainCompleted.allBuilds");
+        body.add("selectFields", "conditionKey");
+        body.add("notificationRecipientType", "de.tum.in.www1.bamboo-server:recipient.server");
```

Comment:
is "de.tum.in.www1.bamboo-server:recipient.server" bound to the Bamboo Server Notification Plugin on https://github.com/ls1intum/bamboo-server-notification-plugin ? I would like to make sure that this migration would also work on Bamboo Systems not hosted at TUM.

Reply:
I'll have to recheck on a local instance as I can't remember if I got the information from a test server or my local instance, but it certainly looks so. Check the pom.xml of the plugin:
https://github.com/ls1intum/bamboo-server-notification-plugin/blob/main/pom.xml#L10-L11

Reply:
Nevermind, no need to check this. See here the config file:
https://github.com/ls1intum/bamboo-server-notification-plugin/blob/main/src/main/resources/atlassian-plugin.xml#L13

And Bamboo creates a `de.tum.in.www1.bamboo-server:recipient.server`